gemelli/ctf.py

```diff
@@ -0,0 +1,179 @@
+import biom
+import skbio
+import qiime2
+from pandas import DataFrame
+from qiime2 import Metadata
+from skbio import OrdinationResults, DistanceMatrix
+from gemelli.factorization import TensorFactorization
+from gemelli.preprocessing import build, rclr
+from gemelli._ctf_defaults import (DEFAULT_COMP, DEFAULT_MSC,
+                                   DEFAULT_MFC, DEFAULT_MAXITER,
+                                   DEFAULT_FMETA as DEFFM)
+
+
+def ctf(table: biom.Table,
+        sample_metadata: qiime2.Metadata,
+        individual_id_column: str,
+        state_column: str,
+        n_components: int = DEFAULT_COMP,
+        min_sample_count: int = DEFAULT_MSC,
+        min_feature_count: int = DEFAULT_MFC,
+        max_iterations_als: int = DEFAULT_MAXITER,
+        max_iterations_rptm: int = DEFAULT_MAXITER,
+        n_initializations: int = DEFAULT_MAXITER,
+        feature_metadata: Metadata = DEFFM) -> (OrdinationResults,
+                                                DistanceMatrix,
+                                                DataFrame,
+                                                DataFrame):
+    # run CTF helper and parse output for QIIME
+    ord_res, dists, straj, ftraj = ctf_helper(table,
```

Comment:
Could you give these the same names as in line 179?

Reply:
they are so long that if I add it this will be a mega block of code.

Reply:
Its your call, but this line is already a bear so a couple characters wouldn't hurt and I think the code would be more intuitive on first glance if the names were not abbreviated. I think the extra length on one line is worth it. You can use convention below for the right side if you need
```
long, tuple_, of_, variables = foo(
    var_one, var_two,
    var_three, var_four)
```